### PR TITLE
docs: translate glossary

### PR DESCRIPTION
**Pull request by ivan-ha**
translate the [glossary page](https://reactjs.org/docs/glossary.html)
maybe @chloewlin can help to label it in #1 ? thanks!

<details>
<summary>Click to view screen capture</summary>

</details>

### CODE REVIEW

**Comment by chloewlin** [comment]
@ivan-ha I labeled it for you!

File: content/docs/reference-glossary.md [hunk 1]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
+React 有兩個不同的方案處理表格輸入。
 
-An input form element whose value is controlled by React is called a *controlled component*. When a user enters data into a controlled component a change event handler is triggered and your code decides whether the input is valid (by re-rendering with the updated value). If you do not re-render then the form element will remain unchanged.
+表格輸入 element 的值是由 React 來控制，就被稱為 *controlled component*。當用戶輸入數據到 controlled component，就會觸發一個 event handler，並且用你的代碼決定輸入是否有效（通過重新 render 更新後的數值）。如果你不重新 render，表格 element 將保持不變。
 
-An *uncontrolled component* works like form elements do outside of React. When a user inputs data into a form field (an input box, dropdown, etc) the updated information is reflected without React needing to do anything. However, this also means that you can't force the field to have a certain value.
+一個 *uncontrolled component* 就像表格 element 一樣在 React 以外工作。當用戶輸入數據到一個表格列（input box，dropdown 等）時，不需要 React 處理任何東西，更新的數據就會被反映出來。但同時這意味著你不能強迫表格列有特定的數值。
 
-In most cases you should use controlled components.
+在大多數的情況下你應用使用 controlled components。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
在大多數的情況下你應用使用被控制的 component。
```

File: content/docs/reference-glossary.md [hunk 2]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
 ## [Keys](/docs/lists-and-keys.html) {#keys}
 
-A "key" is a special string attribute you need to include when creating arrays of elements. Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside an array to give the elements a stable identity.
+「key」是一個特別的 string attribute ，你需要在建立 element array 時加上。Key 幫助 React 分辨那個 element 被更改，添加，或已移除。Key 應該放在 array element 的內部，使 element 有一個穩定的標記。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
「key」是一個特別的 string attribute ，你需要在建立 element array 時加上。Key 幫助 React 分辨那個 element 被更改、增加，或已被移除。Key 應該放在 array element 的內部，使 element 有一個固定的身份。
```

File: content/docs/reference-glossary.md [hunk 3]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
-Keys only need to be unique among sibling elements in the same array. They don't need to be unique across the whole application or even a single component.
+Key 只需要在同一個 array 的 sibling element 中是唯一的。它們不需要在整個應用程式或個別 component 中是唯一。
 
-Don't pass something like `Math.random()` to keys. It is important that keys have a "stable identity" across re-renders so that React can determine when items are added, removed, or re-ordered. Ideally, keys should correspond to unique and stable identifiers coming from your data, such as `post.id`.
+不要將類似 `Math.random()` 的值賦予給 key。Key 有一個誇重新 render 都「隱定的標記」是很重要的，這樣 React 才可以確定項目何時被添加，移除，或重新排序。在理想情況下，key 應該對應來自至於數據唯一而穩定的標記，例如 `post.id`。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
不要將類似 `Math.random()` 的值賦予給 key。重要的是，key 在重新 render 之間具有「固定的身份」，這樣 React 才可以確定何時增加、刪除或重新排序。在理想情況下，key 應該對應來自至於資料唯一而固定的身份，例如 `post.id`。
```

File: content/docs/reference-glossary.md [hunk 4]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
 ## [Refs](/docs/refs-and-the-dom.html) {#refs}
 
-React supports a special attribute that you can attach to any component. The `ref` attribute can be an object created by [`React.createRef()` function](/docs/react-api.html#reactcreateref) or a callback function, or a string (in legacy API). When the `ref` attribute is a callback function, the function receives the underlying DOM element or class instance (depending on the type of element) as its argument. This allows you to have direct access to the DOM element or component instance.
+React 支持一個可以附加到任何 component 的特殊 attribute。Ref attribute 可以是由 [`React.createRef()` function](/docs/react-api.html#reactcreateref) 返回的 object，或一個 callback function，或一個 string (在舊有 API 中)。當 `ref` attribute 是 callback function 時，該 function 接受底層的 DOM element 或 class instance（視乎 element 的類別）作為參數。這使你可以直接訪問 DOM element 或 component instance。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
React 支援一個可以附加到任何 component 的特殊 attribute。Ref attribute 可以是由 [`React.createRef()` function](/docs/react-api.html#reactcreateref) 回傳的 object、或一個 callback function、或一個 string (在舊有 API 中)。當 `ref` attribute 是 callback function 時，該 function 接受底層的 DOM element 或 class instance（取決 element 的類別）作為參數。這讓你可以直接訪問 DOM element 或 component instance。
```

File: content/docs/reference-glossary.md [hunk 5]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
-Use refs sparingly. If you find yourself often using refs to "make things happen" in your app, consider getting more familiar with [top-down data flow](/docs/lifting-state-up.html).
+不要過度使用 ref。假若你發現你在應用程式中經常使用 ref 來「實現某些事情」，請考慮對 [由上而下的數據流](/docs/lifting-state-up.html) 更熟悉。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
不要過度使用 ref。假如你發現在應用程式中經常使用 ref 來「實現某些事情」，請考慮熟悉[由上而下的資料流](/docs/lifting-state-up.html)的概念。
```

File: content/docs/reference-glossary.md [hunk 6]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
 ## [Events](/docs/handling-events.html) {#events}
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
## [Event](/docs/handling-events.html) {#events}
```

File: content/docs/reference-glossary.md [hunk 7]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
 
-Handling events with React elements has some syntactic differences:
+在 React element 中處理 event 有一些語法上的不同：
 
-* React event handlers are named using camelCase, rather than lowercase.
-* With JSX you pass a function as the event handler, rather than a string.
+* React 的 event handler 以 camelCase 來名命，而不是小寫。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
* React 的 event handler 以 camelCase 來命名，而不是小寫。
```

File: content/docs/reference-glossary.md [hunk 8]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
+* 在 JSX 你需要傳入一個 function 到 event handler，而不是一個 string。
 
 ## [Reconciliation](/docs/reconciliation.html) {#reconciliation}
 
-When a component's props or state change, React decides whether an actual DOM update is necessary by comparing the newly returned element with the previously rendered one. When they are not equal, React will update the DOM. This process is called "reconciliation".
+當一個 component 的 props 或 stats 改變時，React 通過比較新返回的 element 和之前 render 的來決定是否需要實際的 DOM 更新。當它們不相等時，React 將更新 DOM。這個過程被稱為「reconciliation」。
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
當一個 component 的 prop 或 state 改變時，React 透過比較新回傳的 element 和之前的  render，來決定是否需要更新實際的 DOM。當它們不相等時，React 將更新 DOM。這個過程被稱為「reconciliation」。
```

File: content/docs/reference-glossary.md [hunk 9]
```diff
@@ -120,47 +120,47 @@ class Welcome extends React.Component {
...
-A "key" is a special string attribute you need to include when creating arrays of elements. Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside an array to give the elements a stable identity.
+「key」是一個特別的 string attribute ，你需要在建立 element array 時加上。Key 幫助 React 分辨那個 element 被更改，添加，或已移除。Key 應該放在 array element 的內部，使 element 有一個穩定的標記。
 
-Keys only need to be unique among sibling elements in the same array. They don't need to be unique across the whole application or even a single component.
+Key 只需要在同一個 array 的 sibling element 中是唯一的。它們不需要在整個應用程式或個別 component 中是唯一。
 
-Don't pass something like `Math.random()` to keys. It is important that keys have a "stable identity" across re-renders so that React can determine when items are added, removed, or re-ordered. Ideally, keys should correspond to unique and stable identifiers coming from your data, such as `post.id`.
+不要將類似 `Math.random()` 的值賦予給 key。Key 有一個誇重新 render 都「隱定的標記」是很重要的，這樣 React 才可以確定項目何時被添加，移除，或重新排序。在理想情況下，key 應該對應來自至於數據唯一而穩定的標記，例如 `post.id`。
 
 ## [Refs](/docs/refs-and-the-dom.html) {#refs}
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
## [Ref](/docs/refs-and-the-dom.html) {#refs}
```

File: content/docs/reference-glossary.md [hunk 10]
```diff
@@ -77,38 +77,38 @@ class Welcome extends React.Component {
 }
 ```
 
-Components can be broken down into distinct pieces of functionality and used within other components. Components can return other components, arrays, strings and numbers. A good rule of thumb is that if a part of your UI is used several times (Button, Panel, Avatar), or is complex enough on its own (App, FeedStory, Comment), it is a good candidate to be a reusable component. Component names should also always start with a capital letter (`<Wrapper/>` **not** `<wrapper/>`). See [this documentation](/docs/components-and-props.html#rendering-a-component) for more information on rendering components. 
+Component 可以分解為不同的功能部件，並在其他 component 中使用。Component 可以返回其他 component，array，string 和 number。一個很好的經驗法則是，如果你的 UI 中有一部份被重複使用多次（Button，Panel，Avatar），或自身就足夠複雜（App，FeedStory，Comment），這些都是成為可重複使用 component 的好選擇。Component 的名字必需以大寫字母開始（`<Wrapper/>` **而不是** `<wrapper/>`）。參見[這裏](/docs/components-and-props.html#rendering-a-component)來了解更多 render component 的資料。
 
 ### [`props`](/docs/components-and-props.html) {#props}
```
Comment (neighborhood999) [inline content/docs/reference-glossary.md]:
```suggestion
### [`prop`](/docs/components-and-props.html) {#props}
```

File: content/docs/reference-glossary.md [hunk 11]
```diff
@@ -1,43 +1,43 @@
...
+這些縮略詞都是指最新版本的 ECMAScript 語言規範標準，而 JavaScript 語言就是它們的一個實現。ES6 版本（也被稱為 ES2015）包含了許多新特性，例如：arrow function，class，樣板字面值，`let` 與 `const` 變數聲明。你可以在[這裏](https://en.wikipedia.org/wiki/ECMAScript#Versions)了解更多版本的新特性。
 
-## Compilers {#compilers}
+## 編譯器 {#compilers}
 
-A JavaScript compiler takes JavaScript code, transforms it and returns JavaScript code in a different format. The most common use case is to take ES6 syntax and transform it into syntax that older browsers are capable of interpreting. [Babel](https://babeljs.io/) is the compiler most commonly used with React.
+JavaScript 編譯器接受 JavaScript 代碼，對其進行轉換並返回不同的格式的 JavaScript 代碼。最常見的使用例子是把 ES6 語法轉換為一些舊瀏覽器能夠辨識的語法。[Babel](https://babeljs.io/) 是 React 上最常用的編譯器。
 
-## Bundlers {#bundlers}
+## 打包工具 {#bundlers}
```
[the remark this answers is not in the document]
Comment (ivan-ha) [reply content/docs/reference-glossary.md]: agree keeping `bundlers` too, would change if no one disagree

File: content/docs/reference-glossary.md [hunk 12]
```diff
@@ -1,43 +1,43 @@
...
 
-Bundlers take JavaScript and CSS code written as separate modules (often hundreds of them), and combine them together into a few files better optimized for the browsers. Some bundlers commonly used in React applications include [Webpack](https://webpack.js.org/) and [Browserify](http://browserify.org/).
+打包工具將多個獨立的 JavaScript 及 CSS 模組（通常有數百個）組合成數個文件，針對瀏覽器作出優化。在 React 應用程式常用的打包工具包括 [Webpack](https://webpack.js.org/) 和 [Browserify](http://browserify.org/)。
```
[the remark this answers is not in the document]
Comment (ivan-ha) [reply content/docs/reference-glossary.md]: will update once confirmed to use `bundlers`

File: content/docs/reference-glossary.md [hunk 13]
```diff
@@ -77,38 +77,38 @@ class Welcome extends React.Component {
...
 
-`props` are inputs to a React component. They are data passed down from a parent component to a child component.
+`props` 是 React component 的輸入。它們是從父 component 傳遞到子 component 的數據。
 
-Remember that `props` are readonly. They should not be modified in any way:
+請記著 `props` 是只讀的。不應該以任何方式來修改它們。
 
 ```js
-// Wrong!
+// 錯誤!
```
[the remark this answers is not in the document]
Comment (ivan-ha) [reply content/docs/reference-glossary.md]: https://github.com/reactjs/zh-hant.reactjs.org/issues/1#issuecomment-465361997

@chloewlin ? should we translate the comments?

**Comment by neighborhood999** [comment]
Hi @ivan-ha, your branch has conflicts, please resolve it, thank you.

**Comment by ivan-ha** [comment]
thanks @neighborhood999 , conflicts resolved

and I updated `打包工具` to `bundlers` as well

**Comment by neighborhood999** [comment]
Hi @ivan-ha, Sorry for late reviewing your update. Can you change `bundlers` to `bundler`? Any question please see the glossary.

**Comment by ivan-ha** [comment]
sure @neighborhood999 , please see latest commit 😄

**Pull request merged**